internal/bminventory/inventory.go

```diff
@@ -6596,12 +6608,13 @@ func (b *bareMetalInventory) updateHostRole(ctx context.Context, host *common.Ho
 	return nil
 }
 
-func (b *bareMetalInventory) updateHostName(ctx context.Context, host *common.Host, hostname *string, db *gorm.DB) error {
+func (b *bareMetalInventory) updateHostName(ctx context.Context, host *common.Host, hostname *string, usages usage.FeatureUsage, db *gorm.DB) error {
 	log := logutil.FromContext(ctx, b.log)
 	if hostname == nil {
 		log.Infof("No request for hostname update for host %s", host.ID)
 		return nil
 	}
+	localhost := host.Host.RequestedHostname == "localhost"
```

Comment:
Same here, so we're setting `from_localhost` to `true` when the hostname _is_ `localhost`?
That seems backwards.